xtask/codegen/js.ungram

```diff
@@ -1238,51 +1239,54 @@ JsImportBareClause =
 // import foo from "mod"
 // import type foo from "mod"
 JsImportDefaultClause =
-		'type'?
-		local_name: AnyJsBinding
-		'from'
-		source: JsModuleSource
-		assertion: JsImportAssertion?
+	'type'?
+	default_specifier: JsDefaultImportSpecifier
+	'from'
+	source: JsModuleSource
+	assertion: JsImportAssertion?
 
 // import * as foo from "mod";
 // import type * as foo from "mod";
 JsImportNamespaceClause =
-  'type'?
-	'*'
-	'as'
-	local_name: AnyJsBinding
+	'type'?
+	namespace_specifier: JsNamespaceImportSpecifier
 	'from'
 	source: JsModuleSource
 	assertion: JsImportAssertion?
 
-// import { a, b: c } from "d"
+// import { a, b: c } from "d";
 //        ^^^^^^^^^^^^^^^^^^^^
-// import c, { b } from "c"
-//        ^^^^^^^^^^^^^^^^^
-// import { type a } from "c"
+// import { type a } from "c";
 //        ^^^^^^^^^^^^^^^^^^^
-// import foo, * as bar from "mod"
-//        ^^^^^^^^^^^^^^^^^^^^^^^^
 // import type { foo } from "mod";
 //        ^^^^^^^^^^^^^^^^^^^^^^^
-// import foo, { type bar } from "mod";
 JsImportNamedClause =
     'type'?
-    default_specifier: JsDefaultImportSpecifier?
-    named_import: AnyJsNamedImport
+	named_specifiers: JsNamedImportSpecifiers
     'from'
     source: JsModuleSource
     assertion: JsImportAssertion?
 
+// import c, { b } from "c"
+//        ^^^^^^^^^^^^^^^^^
+// import foo, { type bar } from "mod";
+//        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+// import foo, * as bar from "mod";
+//        ^^^^^^^^^^^^^^^^^^^^^^^^
+JsImportDefaultExtraClause =
+	default_specifier: JsDefaultImportSpecifier
+	','
+	extra_specifier: AnyJsExtraImportSpecifier
```

Comment:
Is the name "extra" part of the spec? I prefer the name "named" specifier

Reply:
I didn't find a better name...

`named`/`Named` is a bit confusing because it is used for `JsNamedImportSpecifiers`, `AnyJsNamedImportSpecifier`, `JsNamedSpecifier`, `JsShorthandNamedImportSpecifier`, `JsNamedImportSpecifier`, and `JsBogusNamedImportSpecifier`.

Reply:
What about:
```
s/JsImportDefaultExtraClause/JsImportCombinedClause
s/AnyJsExtraImportSpecifier/AnyJsCombinedImportSpecifier
s/extra_specifier/specifier
```

Reply:
I like combined!!